tests/functional/test_s3.py

```diff
@@ -339,6 +339,51 @@ def test_resign_request_with_region_when_needed(self):
                      '?encoding-type=url')
         self.assertEqual(calls[1].url, fixed_url)
 
+    def test_resign_request_in_us_east_1(self):
```

Comment:
It might be good to have the other case when the initial `bad_request_response` should resolve to a 400 on the user side (i.e. bad SSE-C key case). It looks like there is a integration test for this, but it would be nice if there was a functional or unit test as well.

Reply:
Added a test for this case in 68b3997a2b9867ad51a5a590ba511d36f231e149